src/components/fees/HistoricalStakersDistributionList.tsx

```diff
@@ -59,36 +68,102 @@ const tableHeaderCells = [
   },
 ]
 
-const data = [
-  { tokenName: 'vSEY', totalAmount: 7519488.153057938 },
-  { tokenName: 'vKWENTA', totalAmount: 313.37299999999999 },
-  { tokenName: 'vHECO', totalAmount: 2047.43007028 },
-  { tokenName: 'AELIN', totalAmount: 16.846242936861671 },
-]
+enum Status {
+  NotEligible = 'Not eligible',
+  Claimable = 'Claimable',
+  Claimed = 'Claimed',
+}
 
 const HistoricalStakersDistributionList: React.FC = () => {
   const { address: userAddress, appChainId, pushNetwork } = useWeb3Connection()
+  const { isSubmitting, setConfigAndOpenModal } = useTransactionModal()
+
+  const data = [
+    { tokenName: 'vSEY', totalAmount: '7519488153057938939527000' },
+    { tokenName: 'vKWENTA', totalAmount: '313372999999999999999' },
+    { tokenName: 'vHECO', totalAmount: '2047430070280000000000' },
+    { tokenName: 'AELIN', totalAmount: '16846242936861671899' },
+  ]
+
+  const userEntry = useMemo(() => {
+    const item = OptimismDealTokenDistribution.find(
+      (item) => getAddress(item[1] as string) === userAddress,
+    )
+
+    if (!item) {
+      return undefined
+    }
+
+    return {
+      index: item[0] as number,
+      address: item[1] as string,
+      amount: item[2] as string,
+    }
+  }, [userAddress])
+
+  const [isClaimed, refetchIsClaimed] = useContractCall(
+    new JsonRpcProvider(getNetworkConfig(appChainId).rpcUrl),
+    contracts.FEE_DISTRIBUTOR.address[appChainId],
```

Comment:
Is this right? It's using `appChainId` but the fee distribution happens only in OP.

Reply:
It works on Goerli also, I guess we can leave it like that

Reply:
Hmm, I don't think it causes any side effect since if it is not OP, it never gets used. But anyways, it doesn't look correct IMO.

Reply:
Hmm, I don't think it causes any side effect since if it is not OP/Goerli, it never gets used. But anyways, it doesn't look correct IMO.

Reply:
You mean we should throw an error if it's not OP/Goerli? Or what's your suggestion?

Reply:
I would say the logic shoudl be: `if prod -> OP, if dev or testnet -> Goerli`

Reply:
Well, it's not really scalable... I guess we will introduce mainnet very soon

Reply:
Ah I see, it makes sense the. the problem is that your hooks will fail silently in the meantime

Reply:
Ah I see, it makes sense then. the problem is that these hooks will fail silently in the meantime

Reply:
Yeah, I know what you mean... But I don't really see how can we solve this problem properly, I guess it's limitation of React hooks approach